### PR TITLE
[build] Enable streaming telemetry Docker container by default

### PR DESCRIPTION
**Pull request by jleveque**
- Enable streaming telemetry Docker container by default by setting `ENABLE_SYSTEM_TELEMETRY = y` in default config.

- Make src/telemetry/debian/rules executable.

- Add generated files in src/telemetry to .gitignore file

### CODE REVIEW

**Comment by nikos-github** [comment]
Please help answering a couple of things since you are enabling this by default for everyone.

a) What testing has been done? Please include some basic testing etc.
b) What is the default performance impact, if any, on memory and CPU? If impact is measurable, cli would have to be provided to turn on telemetry collection.

**Comment by lguohan** [comment]
agree with nikos.

File: rules/config [hunk 1]
```diff
@@ -69,7 +69,7 @@ ENABLE_ORGANIZATION_EXTENSIONS = y
 #SONIC_PROFILING_ON = y
 
 # ENABLE_SYSTEM_TELEMETRY - build docker-sonic-telemetry for system telemetry support
-# ENABLE_SYSTEM_TELEMETRY = y
+ ENABLE_SYSTEM_TELEMETRY = y
```
Comment (stcheng) [inline rules/config]: one extra space in front of the line

Comment (jleveque) [reply rules/config]: Removed in https://github.com/Azure/sonic-buildimage/pull/2354/commits/c9aeb557bc9a145fc168613de555f9f929f49ff9. Thanks for the catch!

**Comment by lguohan** [comment]
@hui-ma , can you provide answer to nikos question?

**Comment by hui-ma** [comment]
Sure. I agree with Nikos and Guohan too.  @jipanyang has already provided all such info before. 


1. There is auto test suite include in the package: 
https://github.com/Azure/sonic-telemetry/blob/master/doc/grpc_telemetry.md#autotest
We should add it to jenkin auto test for sonic-telemetry repo
"go test -v ./gnmi_server/"

2. The perf test has been done by Jipan and documented here. The cpu load is depending on the polling interval and amount of data is polling. With reasonable value, the cpu load is low: "Without the counter polling activity, the average cpu utilization rate in 60 seconds is 12%. The number changed to 13% with Ethernet ports counters polling every second"
https://github.com/Azure/sonic-telemetry/blob/master/doc/grpc_telemetry.md#performance-and-scale-test

**Comment by nikos-github** [comment]
Is there a way to completely disable the polling through configuration in config_db.json or perhaps configuration under /etc/sonic/telemetry directory? I'm ok with packaging this by default in the image but I need the ability to disable the functionality completely.

**Comment by jleveque** [comment]
@nikos-github: If you want to completely disable the functionality, you can simply add `ENABLE_SYSTEM_TELEMETRY=n` to your build command line. Will this not suffice for you?

**Comment by zhenggen-xu** [comment]
I guess you can just stop telemetry docker.  Linux CLI.  :-)

**Comment by nikos-github** [comment]
@zhenggen-xu That's not how we would like to manage our customers' sonic deployments. We wan't to express customer intent through configuration and not by whacking dockers :-)
Some customers may indeed deploy a script that will stop the docker. Those will be very few, if any at all.

**Comment by jipanyang** [comment]
@nikos-github Being able to enable/disable a docker service while online looks like a requirement from your customer, will you be able to create PR for this?  I think other candidates are snmp, dhcp_relay, and radv.

**Comment by nikos-github** [comment]
@jipanyang Do you have a request to present from someone that is requesting the telemetry functionality that isn't building their own image? If so, lets discuss more in private. To bring this docker in the image by default, it has to come along with a config knob in config_db.json which has to work across reboots and config reloads.

**Comment by jleveque** [comment]
@nikos-github: We have opened an issue to create a knob in ConfigDB with which we can enable/disable the telemetry docker: https://github.com/Azure/sonic-buildimage/issues/2622

**Comment by nikos-github** [comment]
@jleveque When will the knob be ready? By default, the telemetry docker should be disabled. Why did we merge this when there is a valid objection to which Guohan also agreed to?

**Pull request merged**